src/network/protocol/messages/filteradd.zig

```diff
@@ -0,0 +1,78 @@
+const std = @import("std");
+const protocol = @import("../lib.zig");
+
+/// FilterAddMessage represents the "filteradd" message
+///
+/// https://developer.bitcoin.org/reference/p2p_networking.html#filteradd
+pub const FilterAddMessage = struct {
+    element: []const u8,
+
+    const Self = @This();
+
+    pub fn name() *const [12]u8 {
+        return protocol.CommandNames.FILTERADD ++ [_]u8{0} ** 3;
+    }
+
+    /// Returns the message checksum
+    pub fn checksum(self: *const Self) [4]u8 {
+        var digest: [32]u8 = undefined;
+        std.crypto.hash.sha2.Sha256.hash(self.element, &digest, .{});
+        std.crypto.hash.sha2.Sha256.hash(&digest, &digest, .{});
+        return digest[0..4].*;
+    }
+
+    /// Serialize the message as bytes and write them to the Writer.
+    pub fn serializeToWriter(self: *const Self, w: anytype) !void {
+        try w.writeAll(self.element);
+    }
+
+    /// Serialize a message as bytes and return them.
+    pub fn serialize(self: *const Self, allocator: std.mem.Allocator) ![]u8 {
+        const serialized_len = self.hintSerializedLen();
+        const ret = try allocator.alloc(u8, serialized_len);
+        errdefer allocator.free(ret);
+        try self.serializeToSlice(ret);
+        return ret;
+    }
+
+    /// Serialize a message as bytes and write them to the buffer.
+    pub fn serializeToSlice(self: *const Self, buffer: []u8) !void {
+        var fbs = std.io.fixedBufferStream(buffer);
+        try self.serializeToWriter(fbs.writer());
+    }
+
+    pub fn deserializeReader(allocator: std.mem.Allocator, r: anytype) !Self {
```

Comment:
comptime {
            if (!std.meta.hasFn(@TypeOf(w), "readAllAlloc")) @compileError("Expects r to have fn 'readAllAlloc'.");